packages/inference/test/vcr.ts

```diff
@@ -88,45 +87,185 @@ async function vcr(
 		url = input.url;
 	}
 
-	const hash = await hashRequest(url, init);
-	const tapeFile = `./${tapesDirectory}/${hash}.tape.json`;
-	const tapeFileExists = existsSync(tapeFile);
+	const hash = hashRequest(url, init);
+	const tapes = require(TAPES_FILE);
 
 	if (VCR_MODE === "playback") {
-		if (!tapeFileExists) {
-			throw new Error(`Tape not found: ${tapeFile} (${url})`);
+		if (!tapes[hash]) {
+			throw new Error(`Tape not found: ${hash} (${url})`);
 		}
 
-		const rawTape = readFileSync(tapeFile, "utf-8");
-		const tape = JSON.parse(rawTape) as Tape;
+		const response = tapeToResponse(tapes[hash]);
+
+		return response;
+	}
 
-		const response = tapeToResponse(tape);
+	if (VCR_MODE === "cache" && tapes[hash]) {
+		const response = tapeToResponse(tapes[hash]);
 
 		return response;
 	}
 
 	const response = await originalFetch(input, init);
 
-	if (VCR_MODE === "record") {
+	if (VCR_MODE === "record" || VCR_MODE === "cache") {
 		const arrayBuffer = await response.arrayBuffer();
 		const headers: Record<string, string> = {};
 		response.headers.forEach((value, key) => (headers[key] = value));
 
 		const tape: Tape = {
 			url,
-			init,
+			init: {
+				headers: init.headers,
+				method: init.method,
+			},
 			response: {
-				body: Buffer.from(arrayBuffer).toString("base64"),
+				// Truncating the body to 30KB to avoid having huge files
+				body: Buffer.from(arrayBuffer.slice(30 * 1024)).toString("base64"),
 				status: response.status,
 				statusText: response.statusText,
 			},
 		};
+		tapes[hash] = tape;
 
-		writeFileSync(tapeFile, JSON.stringify(tape, null, 2));
+		const { writeFileSync } = await import("node:fs");
+		writeFileSync(`./test/${TAPES_FILE}`, JSON.stringify(tapes, null, 2));
 
 		// Return a new response with an unconsummed body
 		return tapeToResponse(tape);
 	}
 
 	return response;
 }
+
+/**
+ * MD5 implementation in pure JS from https://stackoverflow.com/a/60467595
+ * So we can don't have to care about cross platform issues (Node.js vs browser)
+ */
+function md5(inputString) {
```

Comment:
You can use ```await crypto.subtle.digest("SHA-256", arrayBuffer)```. 

`crypto.subtle` is available both on node 18+ & browser.

Reply:
Yes but then I still need to check if we are in the backend to import `crypto` or just use it directly if we are in the frontend.

Still, I will use the latest API :+1: